admin/services/dfe-signin.service.js

```diff
@@ -6,23 +6,75 @@ const bluebird = require('bluebird')
 const logger = require('../services/log.service').getLogger()
 const jwt = bluebird.promisifyAll(require('jsonwebtoken'))
 const roleService = require('./role.service')
+const schoolDataService = require('./data-access/school.data.service')
+const userDataService = require('./data-access/user.data.service')
+const { MtcHelpdeskImpersonationError } = require('../error-types/mtc-error')
 
-/**
- * @description completes user sign in after DfE callback received.
- * @param {object} user the express request.user object
- * @returns {object} an updated user object with role populated
- */
-const toExport = async (user) => {
-  try {
-    // get role info...
-    const token = await createJwtForDfeApi()
-    const userInfo = await getUserInfoFromDfeApi(token, user)
-    // TODO array check
-    const mtcRole = roleService.mapDfeRoleToMtcRole(userInfo.roles[0].code)
-    user.role = mtcRole
-    return user
-  } catch (error) {
-    throw new Error(`unable to perform post authentication user initialisation:${error.message}`)
+const service = {
+  /**
+   * @description completes user sign in after DfE callback received.
+   * @param {object} user the express request.user object
+   * @returns {object} an updated user object with role populated
+   */
+  getRoleInfo: async (user) => {
+    try {
+      // get role info...
+      const token = await createJwtForDfeApi()
+      const userInfo = await getUserInfoFromDfeApi(token, user)
+      logger.debug('response received from dfe.API:')
+      logger.debugObject(userInfo)
+      // TODO array check
+      const mtcRole = roleService.mapDfeRoleToMtcRole(userInfo.roles[0].code)
+      user.role = mtcRole
```

Comment:
This function takes an object as the param, so it is passed by reference.  I find it slightly misleading that it also returns the object which is then re-assigned by to `req.user` - which is the original object that was passed in.

Reply:
good catch.  i will remove the return and superfluous re-assignment.